onward/app/controllers/RichLinkController.scala

```diff
@@ -1,18 +1,32 @@
 package controllers
 
-import common.{ImplicitControllerExecutionContext, Logging}
+import common.{Edition, ImplicitControllerExecutionContext, JsonComponent, Logging}
 import contentapi.ContentApiClient
 import implicits.Requests
-import model.{ApplicationContext, Content, ContentType}
+import model.{ApplicationContext, Cached, Content, ContentType, Tag}
+import models.dotcomponents.RichLink
 import play.api.mvc.{Action, AnyContent, ControllerComponents, RequestHeader}
 import play.twirl.api.Html
+import views.support.{ImgSrc, Item460, RichLinkContributor}
 
 import scala.concurrent.Future
 
 class RichLinkController(contentApiClient: ContentApiClient, controllerComponents: ControllerComponents)(implicit context: ApplicationContext) extends OnwardContentCardController(contentApiClient, controllerComponents) with Paging with Logging with ImplicitControllerExecutionContext with Requests   {
-
   def render(path: String): Action[AnyContent] = Action.async { implicit request =>
     contentType(path) map {
+        case Some(content) if request.forceDCR =>
+          val richLink = RichLink(
+            content.tags.tags.map(Tag.withoutCommercial), // throw away commercial data as we don't use it
+            content.content.cardStyle.toneString,
+            content.trail.trailPicture.flatMap(tp => Item460.bestSrcFor(tp)),
+            content.trail.headline,
+            content.metadata.contentType,
+            content.content.starRating,
+            content.metadata.commercial.flatMap(_.branding(Edition(request))).map(_.sponsorName),
```

Comment:
Is it worth pulling this and the next line out to give them names and/or using named parameters for RIchLink? At the moment it feels a bit dense/hard to read.

Reply:
good shout, I've updated this